core/unit_test/tstDistributor.hpp

```diff
@@ -625,6 +625,89 @@ void test7( const bool use_topology )
     EXPECT_EQ( slice_dbl_host(0,1), my_rank + 0.5 );
 }
 
+//---------------------------------------------------------------------------//
+void test8( const bool use_topology )
+{
+    // Make a communication plan.
+    std::shared_ptr<Cabana::Distributor<TEST_MEMSPACE> > distributor;
+
+    // Get my rank.
+    int my_rank = -1;
+    MPI_Comm_rank( MPI_COMM_WORLD, &my_rank );
+
+    // Get the comm size.
+    int my_size = -1;
+    MPI_Comm_size( MPI_COMM_WORLD, &my_size );
+
+    // 2 items to the ranks before and after it
+    int num_data = 2;
+    Kokkos::View<int*,TEST_MEMSPACE>
+        export_ranks( "export_ranks", num_data );
+    auto fill_ranks =
+        KOKKOS_LAMBDA( const int )
+        {
+            export_ranks(0) = ( my_rank == 0 ) ? 0 : my_rank - 1;
+            export_ranks(1) = ( my_rank == my_size - 1 ) ? 0 : my_rank + 1;
```

Comment:
Is that a typo?
```suggestion
            export_ranks(1) = ( my_rank == my_size - 1 ) ? my_size - 1 : my_rank + 1;
```

I have a hard time following your comment `"2 items to the ranks before and after it"`

Reply:
This zero actually has to be here to trigger the bug - it creates a condition where rank 0 receives data from a rank it does not send data to.

Reply:
I'll update the comment to make this more clear.